device/iot-device-client/src/main/java/com/microsoft/azure/sdk/iot/device/transport/amqps/AmqpsIotHubConnection.java

```diff
@@ -940,14 +957,25 @@ private AmqpsSessionHandler createSessionHandler(DeviceClientConfig deviceClient
             }
         }
 
+        // If the device session was temporarily unregistered during reconnection, reuse the cached session handler
+        // since it holds the subscription information needed to fully reconnect all links that were open prior to reconnection.
+        for (AmqpsSessionHandler cachedDeviceSessionHandler : this.reconnectingDeviceSessionHandlers)
+        {
+            if (cachedDeviceSessionHandler.getDeviceId().equals(deviceClientConfig.getDeviceId()))
+            {
+                amqpsSessionHandler = cachedDeviceSessionHandler;
```

Comment:
the for loop above has a break statement. Should we add one here?

Reply:
Good idea